pkg/resources/manager/manager.go

```diff
@@ -239,11 +218,24 @@ func (r *resourceManager) allocate(machine *api.Machine, requiredResources core.
 		return err
 	}
 
-	newAvailableResources, err := r.preallocateAvailableResources(requiredResources)
-	if err != nil {
-		return err
+	totalAllocatedRes := core.ResourceList{}
+	for _, s := range r.sources {
+		allocatedRes := s.Allocate(requiredResources)
+
+		sourceResourceList := s.GetAvailableResource()
+		sourceResourceQuantity := sourceResourceList[core.ResourceName(s.GetName())]
+		if sourceResourceQuantity.Sign() == -1 {
+			_ = s.Deallocate(requiredResources)
+			return ErrResourceNotAvailable
+		} else {
+			for k, v := range allocatedRes {
```

Comment:
create additional function for that

Reply:
I was mistaken. It is only used here, so a separate function is not necessary.